torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -214,6 +214,14 @@ auto FunctionParameter::check(PyObject* obj, std::vector<py::handle> &overloaded
       return allow_numbers_as_tensors && THPUtils_checkScalar(obj);
     }
     case ParameterType::SCALAR:
+      if (THPUtils_checkScalar(obj)) {
```

Comment:
Why do we need the extra code here? Is it for readability? The original code does a fallthrough to ParameterType::COMPLEX and then ParameterType::DOUBLE; all of this code seems like it is covered in those two fallbacks

Reply:
It looks like the original idea was for it to fall through, but no that doesn't work. If we reach `ParameterType::Scalar` but the value is an int, we'll fall through to `case ParameterType::Double`, fail the THPUtils_checkDouble, and return false without reaching the `case ParameterType::INT64` case.

Reply:
I don't think that's correct -- `THPUtils_checkDouble` accepts ints: https://github.com/pytorch/pytorch/blob/bf1b4b6feff7be16df2473a54931569465ce0ffc/torch/csrc/utils/python_numbers.h#L102-L114

Reply:
I can try again but I seem to remember this not having worked correctly without the change to prevent falling through to a `return false`. Maybe it was bool or maybe it's not still the case after other changes.

Reply:
I'm not loving this, but it does seem to work.